pipeline_tools/confirm_submission.py

```diff
@@ -2,67 +2,81 @@
 
 import requests
 import argparse
-from .dcp_utils import check_status
-from tenacity import retry, retry_if_result, retry_if_exception_type, wait_exponential, stop_after_delay, RetryError
+from tenacity import retry, retry_if_result, retry_if_exception_type, RetryError, before
+from pipeline_tools.http_requests import HttpRequests
 
-RETRY_SECONDS = 10
-TIMEOUT_SECONDS = 600
 
-
-def status_is_invalid(status):
-    return status != 'Valid'
-
-
-@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS),
-       retry=(retry_if_result(status_is_invalid) | retry_if_exception_type(requests.HTTPError)))
-def wait_for_valid_status(envelope_url):
+def wait_for_valid_status(envelope_url, http_requests):
     """
     Check the status of the submission. Retry until the status is "Valid", or if there is an error with the request to
     get the submission envelope.
 
-    :param str envelope_url: Submission envelope url
-    :return: str status: Status of the submission ("Valid", "Validating", etc.)
-    """
-    print('Getting status for {}'.format(envelope_url))
-    envelope_js = get_envelope_json(envelope_url)
-    status = envelope_js.get('submissionState')
-    print('submissionState: {}'.format(status))
-    return status
+    Args:
+        envelope_url: Submission envelope url
+        http_requests: HttpRequests object
 
+    Returns:
+        Status of the submission ("Valid", "Validating", etc.)
+
+    Raises:
+        requests.HTTPError if response status indicates an error
+    """
+    def log_before(envelope_url):
+        print('Getting status for {}'.format(envelope_url))
+
+    def status_is_invalid(response):
+        envelope_js = response.json()
+        status = envelope_js.get('submissionState')
+        print('submissionState: {}'.format(status))
+        return status != 'Valid'
+
+    response = (
+        http_requests
```

Comment:
Will this write the response.txt for each retried request? In this case, we can expect several retries until the status is `Valid`, so I'm wondering how many files that would create.

Reply:
It could be a lot of files depending on the retry parameters used. I think we should 1) make sure we set the retry parameters wisely so we don't retry an inordinate number of times, which wouldn't serve much purpose 2) only turn recording on when we really need debugging info, such as for scale testing.